.github/workflows/CI-build-test.yml

```diff
@@ -47,6 +43,27 @@ jobs:
       - name: Build
         run: cargo build --release
       - name: Unit tests
-        run: cargo test --lib --bins --all-features --no-fail-fast --release 
+        run: |
+          cargo test --lib --bins --all-features --no-fail-fast --release 2>&1 | tee unit_tests_output.txt
       - name: Integration tests
-        run: cargo test --test '*' --all-features --no-fail-fast --release
+        run: |
+          cargo test --test '*' --all-features --no-fail-fast --release 2>&1 | tee integration_tests_output.txt
+      - name: Upload Unit Test Output
+        uses: actions/upload-artifact@v4
+        with:
+          name: unit-test-output
+          path: unit_tests_output.txt
+          if-no-files-found: warn
+          retention-days: 1
```

Comment:
Is `retention-days: 1` actually working as expected? I was able to download the file today, much more than 1 day has passed since the target run.

Reply:
Interesting! I believe it's implemented as per the docs, but it sounds like something isn't quite right.  I'll keep an eye on it!